File: work/DSA/javascript/timeAndSpaceComplexityAnalysis/findUniqueElement.js
function findUnique(input) {
    let xorValue = 0
    for (let i of input) {
        xorValue ^= i
    }
    return xorValue
}

const readline = require("readline-sync");
let t = Number(readline.question());
while (t--) {
    let size = Number(readline.question());
    let input = [];
    for (let i = 0; i < size; ++i) {
        input.push(Number(readline.question()));
    }
    let answer = findUnique(input)
    console.log(answer)
}

File: work/DSA/c++/timeAndSpaceComplexityAnalysis/findUniqueElement.cpp
#include <bits/stdc++.h>
using namespace std;

int findUnique(int input[], int size)
{

    int xorValue = 0;
    for (int i = 0; i < size; i++)
    {
        xorValue ^= input[i];
    }
    return xorValue;
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t;
    cin >> t;

    while (t--)
    {
        int size;
        cin >> size;
        int *input = new int[size];

        for (int i = 0; i < size; ++i)
        {
            cin >> input[i];
        }

        cout << findUnique(input, size) << endl;
    }
}